### ISSUE TITLE
`nan` not defined for HSV

### ISSUE THREAD

**Issue by Evizero** [JuliaGraphics/ColorVectorSpace.jl#75]
I noticed that `nan` does not work for `HSV`. 

```julia
julia> using Colors, ColorVectorSpace

julia> nan(RGB{Float64})
RGB{Float64}(NaN,NaN,NaN)

julia> nan(HSV{Float64})
ERROR: MethodError: no method matching nan(::Type{ColorTypes.HSV{Float64}})
Closest candidates are:
  nan{T<:AbstractFloat}(::Type{T<:AbstractFloat}) at /home/csto/.julia/v0.5/ColorVectorSpace/src/ColorVectorSpace.jl:46
  nan{C<:Union{ColorTypes.AbstractRGB{T},ColorTypes.Color{T,1},ColorTypes.TransparentColor{C,T,2},ColorTypes.TransparentColor{C,T,4}}}(::Type{C<:Union{ColorTypes.AbstractRGB{T},ColorTypes.Color{T,1},ColorTypes.TransparentColor{C,T,2},ColorTypes.TransparentColor{C,T,4}}}) at /home/csto/.julia/v0.5/ColorVectorSpace/src/ColorVectorSpace.jl:47
```

Is this on purpose? I am asking mainly because the following works

```julia
julia> HSV{Float64}(NaN,NaN,NaN)
HSV{Float64}(NaN,NaN,NaN)
```

**Comment by timholy** [comment]
I think it would be fine, though I'm not sure this is the right package for such definitions. In case it's not clear, the real purpose of this package is to define arithmetic operations for color types for which component-wise vector addition "mostly" makes sense, except for the nonlinearities in the colorspace. XYZ and LMS are genuinely linear vector spaces, (s)RGB is technically not but is universally treated as so in the image processing world (it would be too slow to convert to XYZ, add, and convert back to RGB every time you wanted to add two RGB values).

HSV really doesn't fit that description, for the same reason that `r, θ = r1+r2, θ1 + θ2` does not make sense as a way to add two vectors expressed in polar coordinates. For that reason I've not defined methods here for anything other than Gray, Gray+Alpha, RGB, RGB+Alpha.

We could move the `nan` definitions to ColorTypes, perhaps?

**Comment by Evizero** [comment]
I don't know enough about colors to judge this, but your arguments make sense to me.

(context: I stumbled across this while writing tests for https://github.com/JuliaImages/ImageTransformations.jl/pull/24)

**Comment by johnnychen94** [comment]
Can you explain the idea behind `nan`? Is this a convenient method or is this a deliberately designed method? Why shouldn't `convert(RGB, ::Number)` work?
https://github.com/JuliaGraphics/ColorVectorSpace.jl/blob/371872584639ad9e3415a3826ae655e8bbc5a499/src/ColorVectorSpace.jl#L43-L45

**Comment by timholy** [comment]
> does "reintroduce" indicate that nan once existed in Base and then deprecated?

Yes. In favor of `convert(T, NaN)`.

> Is it because the conversion from Number(or Gray) to Color3 doesn't make sense in general?

That's my feeling, although as you illustrate with `RGB(1,0,0) + Gray(0.2f0)` I guess I haven't been very consistent. I would favor coming to a consistent solution.

**Comment by kimikage** [comment]
I plan to move `nan` to ColorTypes and generalize it for `Colorant`.
One concern is that `nan` is not specific to ColorTypes. In fact, there are packages that do export `nan` 
(e.g. https://github.com/JuliaMath/DoubleFloats.jl).

Since ColorVectorSpace `export`s few types and functions, there is little disadvantage in changing `using ColorVectorSpace` to `import ColorVectorSpace`. On the other hand, ColorTypes `export`s many types, and moreover, they are re-exported by Colors, so not being able to use `using ColorTypes` causes a significant loss of usability.

For this reason, I think it would be better not to export `ColorTypes.nan`, but to allow ColorVectorSpace to export it.

BTW, ColorTypes v0.10.0 or later supports the conversion:
```julia
julia> convert(RGB{Float32}, NaN)
RGB{Float32}(NaN32,NaN32,NaN32)
``` 
However, by extension of this idea, `convert(HSV, NaN)` will still throw an error, and the `alpha` of `convert(ARGB, NaN)` will be `1`. :smile:

**Comment by kimikage** [comment]
BTW, there is a typo in the `nan` test. :ghost:
https://github.com/JuliaGraphics/ColorVectorSpace.jl/blob/0bae88d5fb1c0b328ced1a721257ff9f8e479988/test/runtests.jl#L80-L81

**Comment by kimikage** [comment]
This issue has been solved with the support for ColorTypes v0.11.:tada:
